### ISSUE TITLE
Dynamo-Depth Without ImageNet Pre-Training

### ISSUE THREAD

**Issue by ChristophReich1996** [YihongSun/Dynamo-Depth#19]
Hi @YihongSun,

I'm looking for a fully unsupervised depth estimation model. In particular, without pre-training on ImageNet. I have seen you had an ablation in Tab. 3 of your paper. Would it be possible for you to share the respective checkpoint with me?

Thanks for your help and great work!
Christoph

**Comment by YihongSun** [comment]
Hi @ChristophReich1996,
 
Thank you for your interest in our work!

Unfortunately, I am not able to find the exact model checkpoint that you are referring to (appears to be deleted, sorry! 😔), but hopefully I can share some recent insights on matching performances without ImageNet pretraining.

In our recent [follow-up work](https://arxiv.org/abs/2405.14841), we found that pretraining on random crops of Waymo Open via [MoCov2](https://github.com/facebookresearch/moco) gives very competitive results to pretraining on ImageNet for unsupervised object detection. From that, I believe that MoCov2 pretraining on data patches can directly substitute the original ImageNet backbone without any other changes. 

Since I have not tried this directly, it might not be the case. So if you would like to replicate the training from scratch procedure, please run the following line:
```
python -m torch.distributed.launch --nproc_per_node=4 train.py --cuda_ids 0 1 2 3 -d waymo -n waymo_scratch --epoch_schedules 2 1 5 20 --weights_init scratch
```

I hope this helps and feel free to reply if there's any other questions or concerns!

**Comment by ChristophReich1996** [comment]
Hi @YihongSun,

Thanks for your quick reply! I will try to train from scratch! Have you tried using a ResNet-50 DINO backbone instead of the ImageNet ResNet-18? This seems like a naive fix to keep the performance while being fully unsupervised.

Yeah, I have seen your recent ECCV paper. It's very cool work with very strong results! We also cited your paper in our recent [CVPR paper](https://visinf.github.io/cups/). Are you continuing to work on unsupervised scene understanding? If yes, feel free to shoot me an email!

Cheers,
Christoph

**Comment by ChristophReich1996** [comment]
Maybe a few additional questions: what is the approximate training time on four GPUs of DynamoDepth on Waymo? How stable is the training w.r.t. the number of GPUs (e.g., 8 or 16)?

**Comment by YihongSun** [comment]
Hi @ChristophReich1996,

I have not tried training from scratch with ResNet-50. My guess would be that the performance may be similar to ResNet-18, but a deeper backbone may need more direct supervision to work properly (I am not sure if the standard reconstruction loss is sufficient for that, some pretraining on the target domain may be helpful).

It is awesome to hear that our ECCV was of interest in your work! Your CVPR paper looks very promising with the results are very impressive! I have been recently working on video understanding, and I will definitely reach out if there is any overlap with your research interest!

Regarding training time, depending on your hardware, you should expect the total training time to be around 1 day or so. Please monitor the training as I found training from scratch can be degenerate with the original training schedule (`--epoch_schedules 1 1 5 20`) that assumes a pretrained backbone and I am not sure if `--epoch_schedules 2 1 5 20` is optimal for training from scratch (Sec. 5.2).

**Comment by ChristophReich1996** [comment]
Amazing, thanks a lot! Cheers, Christoph :)

**Comment by YihongSun** [comment]
Hi @ChristophReich1996,

Wrt to the number of GPUs, I didn't try training with more than 4 gpus, but typically 4 is much more stable than 1 or 2 and I didn't observe any degenerations when using 4 gpus.

**Comment by ChristophReich1996** [comment]
Hi @YihongSun,

Thanks again for your help! We were able to reproduce the results using a DINO ResNet-18 on Cityscapes :) Great work!
